wordpress-mobile/WordPress-Android
# Issue/add unified comments feature flag

This PR adds feature flag and an entry point to the comment list for Unified Comment experience project.

In addition, legacy classes are deprecated with an explanatory message.

[![Image from Gyazo](https://i.gyazo.com/9cb38e2a4c113f80be10670179823718.png)](https://gyazo.com/9cb38e2a4c113f80be10670179823718)

Note: I added a detekt suppress method [here](https://github.com/wordpress-mobile/WordPress-Android/pull/14679/files#diff-9dfb54ffb216bc52a271e3148a25e73b0dbb41ca52e5cf0fba909f984e4cadaeR16) since the feature flag is temporary.

To test:
- Install `vanilla` variant of the app, and confirm that you do not see Unified Comments item on MySite screen.
- Install `wasabi` variant of the app, and confirm that you DO see Unified Comments, and clicking on it opens an empty screen.

Test with ImprovedMySiteFragment:
- Add `buildConfigField "boolean", "MY_SITE_IMPROVEMENTS", "true"` to `wasabi` build variant and install the app.
- Confirm that you DO see Unified Comments, and clicking on it opens an empty screen.

## Regression Notes
1. Potential unintended areas of impact
- Unified Comment row is visible in Vanilla variant.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual and automatic testing.

3. What automated tests I added (or what prevented me from doing so)
- Added couple of automated tests to code that powers `ImprovedMySiteFragment`
PR submission checklist:
